File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        unordered_map<string, vector<string>> m;
        vector<vector<string>> res;
        const char SLASH = '/';
        const char SPACE = ' ';
        for(auto& p : paths) {
            stringstream ss(p);
            string path;
            string file;
            getline(ss, path,  SPACE);
            while(getline(ss, file, SPACE)) {
                m[file.substr(file.find('('))].push_back(path + SLASH + file.substr(0, file.find('(')));
            }
        }
        for(auto& e : m) {
            if(e.second.size() != 1)
                res.push_back(e.second);
        }

        return res;
    }
};

File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        
        a = []
        re = dict()
        result = []
        
        for path in paths:
            a.append(list(path.split()))
            
        
        for files in a:
            directory = files[0]
            for file in files[1:]:
                filename, content = file.split( '(' )
                content = content[:-1]
                new_file_path = directory+'/'+filename
                if content in re :
                    re[content].append(new_file_path)
                else:
                    re[content] = [new_file_path]
      
        return list(filter(lambda elem: len(elem) > 1, list(re.values())))